test/elm/interval/data.cql

```diff
@@ -374,6 +374,68 @@ define PosInfEndNotBeforeDateIvl: Interval[DateTime(2013, 1, 1, 0, 0, 0, 0), nul
 define UnknownEndMayBeBeforeDateIvl: Interval[DateTime(2013, 1, 1, 0, 0, 0, 0), null) before Interval[DateTime(2020, 1, 1, 0, 0, 0, 0), DateTime(2040, 1, 1, 0, 0, 0, 0)]
 define UnknownEndNotBeforeDateIvl: Interval[DateTime(2013, 1, 1, 0, 0, 0, 0), null) before Interval[DateTime(2000, 1, 1, 0, 0, 0, 0), DateTime(2013, 1, 1, 0, 0, 0, 0)]
 
+// @Test: BeforeOrOn
```

Comment:
It might be nice to add a few more test cases to exercise these components of the spec (for both `BeforeOrOn` and `AfterOrOn`):
1. Mixed `Date` and `DateTime` intervals
> When this operator is called with a mixture of Date- and DateTime-based intervals, the Date values will be implicitly converted to DateTime values as defined by the ToDateTime operator.
2. Mixed point values and intervals:
> ```
> same _precision_ or before(left T, right Interval<T>) Boolean
> same _precision_ or before(left Interval<T>, right T) Boolean
> ```
> When this operator is called with a mixture of point values and intervals, the point values are implicitlyconverted to an interval starting and ending on the given point value.

In both these cases, CQL-to-ELM does the heavy lifting (adding the implicit conversions), but it still might be good to have one or two test-cases to prove the point and show that this behavior is in fact covered.

Reply:
As for priority: This is a nice-to-have.  If you do not have the time to do it, I will still approve the PR.

Reply:
Added tests for these situations in 4fb28c1738badcd62aaa2aab068e8e3e3b0d8960.